### DISCUSSION TITLE
Better routing system

### DISCUSSION THREAD

**Discussion by andim27** [nylo-core/nylo#88] [Ideas] [answered]
Need the ability to get the previous route name
Something like this:` NyRouter().getPreviousRouteName()`
May be we need store router-history:` NyRouter.instance.routeHistory.` In this case we need have `const maxRouterHistory = 20`(may be less or more).It's very useful to know from what page(routerPath) was call

**Comment by agordn52** [comment]
Hi @andim27,

Yeah, a route history would be nice.
It could just save to the instance as you mentioned like `routeHistory`.

I'll do some testing.

**Reply by andim27** [reply]
Great! Let's make Nylo better!))

**Reply by agordn52** [reply, marked as answer]
Hi @andim27,

In the latest version, you can now fetch the routeHistory and more 🚀

Here are some new features in [5.15.0](https://pub.dev/packages/nylo_framework):
`Nylo.getRouteHistory() // Get the route history` 
`Nylo.getCurrentRoute() // Get the current route`
`Nylo.getPreviousRoute() // Get the previous route`
`Nylo.getCurrentRouteName() // Get the current route name`
`Nylo.getPreviousRouteName() // Get the previous route name`
`Nylo.getCurrentRouteArguments() // Get the current route arguments`
`Nylo.getPreviousRouteArguments() // Get the previous route arguments using`

Here's the upgrade guide from v5.13.0
https://github.com/nylo-core/nylo/compare/v5.13.0...v5.14.0

Hope that helps.

**Reply by andim27** [reply]
Thank's! It's time to upgrade to last Nylo!))
